packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -383,35 +423,42 @@ class ReactShallowRenderer {
       'Too many re-renders. React limits the number of renders to prevent ' +
         'an infinite loop.',
     );
+    let updatesMap;
 
     if (componentIdentity === this._currentlyRenderingComponent) {
       // This is a render phase update. Stash it in a lazily-created map of
       // queue -> linked list of updates. After this render pass, we'll restart
       // and apply the stashed updates on top of the work-in-progress hook.
       this._didScheduleRenderPhaseUpdate = true;
+      if (this._renderPhaseUpdates === null) {
+        this._renderPhaseUpdates = new Map();
+      }
+      updatesMap = this._renderPhaseUpdates;
+    } else if (componentIdentity === this._previousComponentIdentity) {
+      // This means an update has happened after the function component has
```

Comment:
Should it also rerender in this case?

It would be inline with class component behavior, covering tests like
```js
rendered.props.onClick();
expect(shallowRenderer.getRenderOutput()).toEqual(... updated ...);
```
which is what I think a lib like enzyme would do internally

Reply:
I'm not sure how Enzyme works, but in this case, it's triggering an update to the update queue. There is no re-render, which seems more in line with how shallow renderer works (you explicitly call `render`).

Reply:
But don't shallow-render `Updater` do this already? (immediately change the state and rerender)
https://github.com/facebook/react/blob/master/packages/react-test-renderer/src/ReactShallowRenderer.js#L154-L159

Reply:
Yeah setState _should_ trigger a render.

Reply:
Ah, my mistake. I've changed the PR to do a re-render.